tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/field/field_datatype_test.dart

```diff
@@ -1032,6 +1033,176 @@ void main() {
     );
   });
 
+  group('Given a class with a type set to the class name of a custom type', () {
+    var type = TypeDefinition(
+      className: 'CustomExample',
+      generics: const [],
+      nullable: false,
+      url: 'package:shared_package/src/lib/custom_example.dart',
+      customClass: true,
+    );
+
+    var config = GeneratorConfigBuilder().withExtraClasses([type]).build();
+    var models = [
+      ModelSourceBuilder().withYaml(
+        '''
+          class: Example
+          fields:
+            name: CustomExample
+          ''',
+      ).build()
+    ];
+
+    var collector = CodeGenerationCollector();
+    StatefulAnalyzer analyzer = StatefulAnalyzer(
+      config,
+      models,
+      onErrorsCollector(collector),
+    );
+    var definitions = analyzer.validateAll();
+
+    test('then no errors was generated', () {
+      expect(
+        collector.errors,
+        isEmpty,
+        reason: 'Expected no errors, but one was generated.',
+      );
+    });
+
+    test('then the field type is set.', () {
+      var definition = definitions.first as ClassDefinition;
+      expect(definition.fields.first.type.className, 'CustomExample');
+    });
+
+    test('then the type url is set to the custom type.', () {
+      var definition = definitions.first as ClassDefinition;
+      expect(
+        definition.fields.first.type.url,
+        'package:shared_package/src/lib/custom_example.dart',
+      );
+    });
+  });
+
+  group(
+      'Given a class with a type set to a list of the class name of a custom type',
```

Comment:
Suggestion(Non blocking): Change to: `Given a class with a type set to a list of custom classes`

Reply:
Suggestion for the same name change for the other map test.